Speech analysis code updates

jmculnan opened the pull request:
Speech analysis code updated to: 
1. Get the probability distribution over prediction label space
2. Gather penultimate layer values of neural network
3. Accept updated json file format (words + sentence json together, multiple json per file separated by a blank line) and return the probability of each label in the space and the penultimate layer in output json

There is now redundant code (appearing in both `tomcat_speech/train_and_test_models/asist_analysis_functions.py` and `scripts/run_asist_analysis`)

adarshp reviewed (changes requested):
Almost there! Could you please make the following changes:

- the `get_json_output_of_speech_analysis` function takes arguments corresponding to input and output JSON files. However, in order to integrate this code by either (i) having the speechAnalyzer program communicate with a 'serverized' version of this over websockets or using HTTP requests, or (ii) having Vincent rewrite this in C++ using the Torch C++ API for more direct integration with speechAnalyzer, it will be easiest to have a pure function (i.e. one that does no I/O) to hook into (or translate to C++).

jmculnan commented:
@adarshp I've updated the code in the following ways: 

1.  altered `get_json_output_of_speech_analysis` (starts l. 43) so that file reading and writing takes place in separate, smaller functions (`read_in_aligned_json` and `save_json_predictions`, respectively)
2. removed redundant code from `run_asist_analysis`
